crates/core/src/client/api.rs

```diff
@@ -54,12 +56,21 @@ pub trait KakarotEthApi<P: Provider + Send + Sync>: KakarotStarknetApi<P> + Send
         block_id: BlockId,
     ) -> Result<U256, EthApiError<P::Error>>;
 
+    async fn token_allowance(
```

Comment:
`token_allowance`, `token_balances` and `token_metadata` don't actually belong to the Eth API, what do you think of making a new trait for Alchemy methods?

Reply:
Do you have any preferences? The logic is almost the same as in the `U256` case so I'm wondering how different the new error variant should be.

Reply:
I initially started moving over the methods to a new trait implementation but this involved updating how the RpcModule is built at the top and I was getting into some messy logic. Specifically, in `eth-rpc/src/rpc.rs` we would have to add another argument to the `new` method that takes a `KakarotAlchemyApi` type and pass that into the `AlchemyRpc::new()` call. Not sure how you'd want this handled.